src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/Serialization/ProjectSnapshotHandleJsonConverter.cs

```diff
@@ -35,7 +33,7 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                     case nameof(ProjectSnapshotHandle.FilePath):
                         if (reader.Read())
                         {
-                            filePath = (string)reader.Value;
+                            filePath = (string)reader.Value!;
```

Comment:
This is 100% `Newtonsoft.Json` issue. The `Read` method should have an attribute to tell the compiler that `Value` is not null if it returned `true`.

Reply:
+1 argument for moving to `System.Text.Json` btw 🙂